ui/src/components/LocationFieldset_test.js

```diff
@@ -0,0 +1,252 @@
+  beforeEach(() => {
+    global.ENV = {
+      'maps_api_key': 'abc123',
+    };
+    jest.mock('./../utils/Utils');
+    mockLoadExternalScript = jest.fn();
+    Utils.loadExternalScript = mockLoadExternalScript.bind(Utils);
+    global.google = mockGoogle;
+  });
+
+  afterEach(() => {
+    clearConstructedMaps();
+    clearConstructedMarkers();
+  });
+
+  test('location text should be populated from props', () => {
+    const wrapper = mountWithIntl(
+      <LocationFieldset
+          mapDefaultCenter={[40.6782, -73.9442]}
+          mapDefaultZoom={10}
+          locationText='burlington, vt' />
+    );
+    wrapper.update();
+    expect(wrapper.find('Input').get(0).props.value).toBe('burlington, vt');
+    wrapper.unmount();
+  });
+
+  test('location text update should call back', () => {
+    const mockLocationTextUpdateCallback = jest.fn();
+    const wrapper = mountWithIntl(
+      <LocationFieldset
+          mapDefaultCenter={[40.6782, -73.9442]}
+          mapDefaultZoom={10}
+          onLocationTextUpdate={mockLocationTextUpdateCallback} />
+    );
+    const changeEvent = {target: {value: 'burlington, vt'}};
+    wrapper.find('Input').simulate('change', changeEvent);
+    wrapper.update();
+    expect(mockLocationTextUpdateCallback).toHaveBeenCalledWith(
+        'burlington, vt');
+    wrapper.unmount();
+  });
+
+  test('no map button present without Maps API key', () => {
+    global.ENV = {};
+    const wrapper = mountWithIntl(
+        <LocationFieldset
+            mapDefaultCenter={[40.6782, -73.9442]}
+            mapDefaultZoom={10}
+            />);
+    wrapper.update();
+    expect(wrapper.find('button').length).toBe(1);
+    wrapper.unmount();
+  });
+
+  test('no map button present with empty Maps API key', () => {
+    global.ENV = {'maps_api_key': ''};
+    const wrapper = mountWithIntl(
+        <LocationFieldset
+            mapDefaultCenter={[40.6782, -73.9442]}
+            mapDefaultZoom={10}
+            />);
+    wrapper.update();
+    expect(wrapper.find('button').length).toBe(1);
+    wrapper.unmount();
+  });
+
+  test('map button present with Maps API key', () => {
+    const wrapper = mountWithIntl(
+        <LocationFieldset
+            mapDefaultCenter={[40.6782, -73.9442]}
+            mapDefaultZoom={10}
+            />);
+    wrapper.update();
+    expect(wrapper.find('button').length).toBe(2);
+    expect(wrapper.find('button').at(1).text()).toBe('Show map');
+    wrapper.unmount();
+  });
+
+  test('map script is loaded on show map button click', () => {
+    const wrapper = mountWithIntl(
+        <LocationFieldset
+            mapDefaultCenter={[40.6782, -73.9442]}
+            mapDefaultZoom={10}
+            />);
+    wrapper.update();
+    wrapper.find('button').at(1).simulate('click');
+    expect(mockLoadExternalScript).toHaveBeenCalledWith(
+        'https://maps.googleapis.com/maps/api/js?key=abc123',
+        expect.anything());
+    wrapper.unmount();
+  });
```

Comment:
Add a test that the map script is not loaded twice if the user shows the map twice?

Reply:
Done.